phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -285,17 +286,25 @@ private void logToIndexToolOutputTable(byte[] dataRowKey, byte[] indexRowKey, lo
                                            String errorMsg) {
         try (Table hTable = ServerUtil.ConnectionFactory.getConnection(ServerUtil.ConnectionType.INDEX_WRITER_CONNECTION,
                 env).getTable(TableName.valueOf(IndexTool.OUTPUT_TABLE_NAME))) {
-            byte[] rowKey = new byte[Long.BYTES];
-            Bytes.putLong(rowKey, 0, scan.getTimeRange().getMax());
+            byte[] rowKey;
+            if (dataRowKey != null) {
+                int length = Long.BYTES + dataRowKey.length;
+                rowKey = new byte[length];
+                Bytes.putLong(rowKey, 0, scan.getTimeRange().getMax());
+                for (int i = Long.BYTES, j = 0; i < length; i++, j++) {
```

Comment:
Nit: System.arraycopy() might perform better than byte-by-byte copy. I assume it's java equivalent of memcpy.

Reply:
See also Bytes.putBytes

Reply:
Both suggestions work. I will go with Bytes.putBytes as it makes the code more readable.